lib/uri/generic.rb

```diff
@@ -659,7 +659,7 @@ def hostname
     # it is wrapped with brackets.
     #
     def hostname=(v)
-      v = "[#{v}]" if /\A\[.*\]\z/ !~ v && /:/ =~ v
+      v = "[#{v}]" if v.index(':') && v[0] != '[' && v[-1] != ']'
```

Comment:
For the record, I have measured that `'[' + v + ']'` is actually a bit slower than interpolation. So keeping this as-is.